client/modules/workspace/src/AppsBreadcrumb/AppsBreadcrumb.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from 'react';
+import { Breadcrumb } from 'antd';
+import { Link, useParams, useLocation } from 'react-router-dom';
+import styles from './AppsBreadcrumb.module.css';
+import { TAppParamsType, TAppResponse, useGetApps } from '@client/hooks';
+
+function getPathRoutes(path: string = '') {
+  const pathComponents = decodeURIComponent(path)
+    .split('/')
+    .filter((p) => !!p);
+
+  return pathComponents.map((comp, i) => ({
+    title: comp,
+    path: '/' + encodeURIComponent(pathComponents.slice(0, i + 1).join('/')),
+  }));
+}
+
+export const AppsBreadcrumb: React.FC = () => {
+  const [appData, setAppData] = useState<TAppResponse | null>(null);
+  const { appId } = useParams() as TAppParamsType;
+  const appVersion = new URLSearchParams(useLocation().search).get(
+    'appVersion'
+  ) as string | undefined;
+
+  const { data, isLoading } = useGetApps({ appId, appVersion });
+
+  useEffect(() => {
+    if (data) {
+      setAppData(data);
+    }
+  }, [data]);
+
+  const { pathname } = useLocation();
+  const breadcrumbItems = [
+    { title: 'Home', path: window.location.origin },
+    { title: 'Use DesignSafe' },
+    { title: 'Tools & Applications', path: '/' },
+  ];
+
+  // Modify the path for Job Status
+  let modifiedPath = pathname;
+  if (pathname.endsWith('/history')) {
+    modifiedPath = 'Job Status';
+  } 
+
+  return (
+    <div className={styles.breadcrumbWrapper}>
+      {!isLoading && (
+        <Breadcrumb
+          className={styles.appsBreadcrumb}
+          items={[...breadcrumbItems, ...getPathRoutes(modifiedPath)]}
+          itemRender={(obj) => {
+            if (!obj.path) {
+              return <span className="breadcrumb-text">{obj.title}</span>;
+            }
+            let title = obj.title;
+            if (appData && obj.title !== 'Home' && obj.title !== 'Tools & Applications' && obj.title !== 'Job Status') {
+              title = appData.definition.notes?.label || appData.definition.id || obj.title;
+            }
+            return (
+              <Link className="breadcrumb-link" to={obj.path}>
```

Comment:
If the element is the last item in the list (i.e. currently loaded), we can remove the link part of the element